--- lcms-1.17/include/lcms2.h	2007-07-27 12:00:01.000000000 +0200
+++ lcms-1.17/include/lcms2.h-new	2008-03-28 19:02:33.000000000 +0100
@@ -47,7 +47,10 @@
 // ********** Configuration toggles ****************************************
 
 // Uncomment this one if you are using big endian machines
-// #define CMS_USE_BIG_ENDIAN   1
+#include <endian.h>
+#if __BYTE_ORDER == __BIG_ENDIAN
+#define USE_BIG_ENDIAN   1
+#endif
 
 // Uncomment this one if your compiler/machine does NOT support the
 // "long long" type.
