Index: Pillow-2.1.3/setup.py
===================================================================
--- Pillow-2.1.3.orig/setup.py
+++ Pillow-2.1.3/setup.py
@@ -282,12 +282,6 @@
             if os.path.isfile(os.path.join(tcl_dir, "tk.h")):
                 _add_directory(include_dirs, tcl_dir)
 
-        # standard locations
-        _add_directory(library_dirs, "/usr/local/lib")
-        _add_directory(include_dirs, "/usr/local/include")
-
-        _add_directory(library_dirs, "/usr/lib")
-        _add_directory(include_dirs, "/usr/include")
 
         #
         # insert new dirs *before* default libs, to avoid conflicts
