--- stm-cross-binutils.spec.orig	2013-05-31 19:00:00.611319283 +0200
+++ stm-cross-binutils.spec	2013-05-31 19:00:00.611319283 +0200
@@ -23,6 +23,7 @@
 # SH4 patches
 Patch100: binutils-2.23.2-stm-130806.patch
 Patch101: binutils-2.23.2-config_bfd.patch
+Patch102: stm-cross-binutils.%{version}-%{release}.diff
 
 # ST200 patches
 Patch200: binutils-2.21.51-st200-7.03-6-0-0-B-111108.patch
@@ -37,9 +38,6 @@
 BuildRequires: texinfo
 %endif
 BuildRequires: %{_stm_pkg_prefix}-host-autoconf
-BuildRequires: %{_stm_pkg_prefix}-host-bison
-BuildRequires: %{_stm_pkg_prefix}-host-flex
-BuildRequires: %{_stm_pkg_prefix}-host-gettext
 BuildRequires: %{_stm_pkg_prefix}-host-rpmconfig
 
 Group: Development/Languages
@@ -90,6 +88,7 @@
 %if %target_cpu sh
 %patch100 -p1
 %patch101 -p1
+%patch102 -p1
 %endif
 
 %if %target_cpu st200
@@ -117,6 +116,7 @@
 %cross_configure \
        --enable-nls \
        --disable-multilib \
+       --disable-werror \
        --with-system-zlib \
 %if "%{_stm_target_fpu}" == "yes"
 %if %target_arch armv7 armv7_uclibc
@@ -143,10 +143,6 @@
 
 %make
 
-%if %target_cpu sh
-(cd bfd ; make headers)
-%endif
-
 %install
 %cross_setup
 srcdir=$PWD
