--- stm-target-sysvinit.spec.orig
+++ stm-target-sysvinit.spec
@@ -19,6 +19,7 @@
 Patch6		: sysvinit-2.86-environ.patch
 Patch7		: sysvinit-mountvirtfs.patch
 Patch8		: sysvinit-mountvirtfs-debugfs.patch
+Patch101		: stm-target-sysvinit-2.86-15.patch
 License 	: GPL
 Group	  	: System Environment/Base
 Buildroot 	: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-%{_stm_target_arch}-XXXXXX)
@@ -89,6 +89,7 @@
 %patch6 -p1
 %patch7 -p1
 %patch8 -p1
+%patch101 -p1
 
 # tidy up the .orig files as we use wildcards later
 find . -name "*.orig" -print | xargs -r rm
