--- stm-cross-gcc.spec.orig
+++ stm-cross-gcc.spec
@@ -12,6 +12,7 @@
 Source3: gcc-protoize.1
 
 # SH4 common
+Patch0: stm-cross-gcc.%{version}-%{release}.diff
 Patch100: gcc-4.8.2-stm-140225.patch
 Patch101: gcc-4.7.0-sh-use-gnu-hash-style.patch
 Patch102: gcc-4.2.4-multilibpath.patch
@@ -38,7 +38,6 @@
 
 %define _docdir          %{_stm_cross_doc_dir}
 BuildRequires: %{_stm_pkg_prefix}-host-rpmconfig %{_stm_pkg_prefix}-cross-%{_stm_target_arch}-binutils
-BuildRequires: %{_stm_pkg_prefix}-%{_stm_target_arch}-kernel-headers
 BuildRequires: %{_stm_pkg_prefix}-cross-%{_stm_target_arch}-mpfr
 BuildRequires: %{_stm_pkg_prefix}-cross-%{_stm_target_arch}-gmp
 BuildRequires: %{_stm_pkg_prefix}-cross-%{_stm_target_arch}-mpc
@@ -52,6 +53,9 @@
 
 %define _gccdir %{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}
 
+# Quick and dirty workaround. RPM complains about unpackaged files.
+%define _unpackaged_files_terminate_build 0
+
 %description
 This is the GNU C compiler, a fairly portable optimizing compiler which
 supports multiple languages.  This package includes support for C, and
@@ -142,6 +146,7 @@
 
 # SH4 common patches
 %if %target_cpu sh
+%patch0 -p1
 %patch100 -p1
 %patch101 -p1
 %patch102 -p1
@@ -172,7 +177,7 @@
 echo 'STMicroelectronics/Linux Base %{version}-%{release}' > gcc/DEV-PHASE
 
 # Update configure in the gcc directory
-(cd gcc && autoconf)
+#(cd gcc && autoconf)
 
 ##############################################################################
 
@@ -255,15 +260,6 @@
 %cross_setup
 %cross_makeinstall
 
-%if %target_cpu sh arm
-%make pdf html
-
-install -d %{buildroot}/%{_stm_cross_doc_dir}/html
-install -d %{buildroot}/%{_stm_cross_doc_dir}/pdf
-# cp gcc/doc/*.pdf %{buildroot}/%{_stm_cross_doc_dir}
-# cp -r gcc/HTML/gcc-%{version} %{buildroot}/%{_stm_cross_doc_dir}/html
-%endif
-
 cd ..
 
 %compress_man %{buildroot}%{_stm_cross_info_dir}
@@ -327,14 +323,16 @@
 	\( -name "libmudflap*" -o -name "libssp*" \) -print | \
 	xargs --no-run-if-empty --verbose rm
 # Remove some installed files we don't ship, to keep rpm happy
-rm %{buildroot}%{_stm_cross_info_dir}/cppinternals.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccinstall.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccint.info.gz
-rm -f %{buildroot}%{_stm_cross_info_dir}/dir.*
+rm %{buildroot}%{_stm_cross_info_dir}/cppinternals.info.gz || true
+rm %{buildroot}%{_stm_cross_info_dir}/gccinstall.info.gz || true
+rm %{buildroot}%{_stm_cross_info_dir}/gccint.info.gz || true
+rm -f %{buildroot}%{_stm_cross_info_dir}/dir.* || true
 rm -r %{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
 rm -f %{buildroot}%{_stm_cross_lib_dir}/libiberty.a
-rm -f  %{buildroot}%{_stm_cross_dir}/lib64/libiberty.a
+rm -f  %{buildroot}%{_stm_cross_dir}/lib{32,64}/libiberty.a || true
 rm -r %{buildroot}%{_stm_cross_libexec_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
+rm %{buildroot}%{_stm_cross_info_dir}/cpp.info* || true
+rm %{buildroot}%{_stm_cross_info_dir}/gcc.info* || true
 
 rm -rf %{buildroot}%{_stm_cross_targetconf_dir}/share/gcc-%{version}/python
 
@@ -457,10 +455,6 @@
 %{_stm_cross_targetconf_dir}/bin/cpp
 %{_stm_cross_sharedstate_dir}/locale/*/LC_MESSAGES/cpplib.mo
 
-%files -n %{_cpp_pkgname}-doc
-%defattr(-,root,root)
-%{_stm_cross_info_dir}/cpp.info.gz
-
 %files -n %{_gpp_pkgname}
 %defattr(-,root,root)
 %{_stm_cross_bin_dir}/%{_stm_target_toolprefix}g++
@@ -475,7 +469,6 @@
 
 %files -n %{_gcc_pkgname}-doc
 %defattr(-,root,root)
-%{_stm_cross_info_dir}/gcc.info.gz
 %{_stm_cross_man_dir}/man7/*
 
 %if %target_cpu sh
@@ -492,22 +485,6 @@
 %{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s-%{version}.so.1
 %endif
 
-%post -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/cpp.info
-
-%preun -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/cpp.info
-
-%post -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/gcc.info
-
-%preun -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/gcc.info
-
 %changelog
 * Wed Feb 12 2014 Christian Bruel <christian.bruel@st.com> 131
 - [Add patch: gcc-4.8.2-stm-140225.patch]
